yb-voyager/cmd/config.go

```diff
@@ -0,0 +1,467 @@
+	2. Ensures that all section names (e.g., export-schema) are known and valid.
+	3. Ensures that all nested keys inside each section are valid for that section.
+	4. Ensures that only one section from each group of mutually exclusive aliases is used (e.g.,
+	"export-data" vs "export-data-from-source").
+
+	Any invalid global keys, unknown sections, or invalid section keys are collected and printed.
+	Conflicts between mutually exclusive sections are also reported.
+	If any validation error is found, an error is returned.
+
+	This helps catch misconfigurations early and provides comprehensive feedback to the user.
+*/
+func validateConfigFile(v *viper.Viper) error {
+	var (
+		invalidGlobalKeys   []string
+		invalidSectionKeys  = make(map[string][]string)
+		invalidSections     []string
+		conflictingSections = [][]string{}
+		presentSections     = make(map[string]bool)
+	)
+
+	fmt.Println("Validating config file...")
+	for _, key := range v.AllKeys() {
+		parts := strings.Split(key, ".")
+		if len(parts) == 1 {
+			// Check global level keys
+			if !allowedGlobalConfigKeys[key] {
+				invalidGlobalKeys = append(invalidGlobalKeys, key)
+			}
+		} else {
+			// Validate section-based keys
+			section, nestedKey := parts[0], parts[1]
+			presentSections[section] = true
+			if allowedConfigSections[section] == nil {
+				// Unknown section
+				if !utils.ContainsAnyStringFromSlice(invalidSections, section) {
+					invalidSections = append(invalidSections, section)
+				}
+			} else if !allowedConfigSections[section][nestedKey] {
+				// Invalid key inside a known section
+				invalidSectionKeys[section] = append(invalidSectionKeys[section], nestedKey)
+			}
+		}
+	}
+
+	// Check for mutually exclusive section usage
+	for _, group := range aliasCommandsPrefixes {
+		var used []string
+		for _, sec := range group {
+			if presentSections[sec] {
+				used = append(used, sec)
+			}
+		}
+		if len(used) > 1 {
+			conflictingSections = append(conflictingSections, used)
+		}
+	}
+
+	// If invalid configurations exist, print them
+	if len(invalidGlobalKeys) > 0 || len(invalidSectionKeys) > 0 || len(invalidSections) > 0 || len(conflictingSections) > 0 {
+		if len(invalidGlobalKeys) > 0 {
+			fmt.Printf("%s %v\n", color.RedString("Invalid global config keys:"), invalidGlobalKeys)
+		}
+		for section, keys := range invalidSectionKeys {
+			printStatement := fmt.Sprintf("Invalid keys in section '%s':", section)
+			fmt.Printf("%s %v\n", color.RedString(printStatement), keys)
+		}
+		if len(invalidSections) > 0 {
+			fmt.Printf("%s %v\n", color.RedString("Invalid sections:"), invalidSections)
+		}
+		if len(conflictingSections) > 0 {
+			for _, conflict := range conflictingSections {
+				fmt.Printf("%s %v\n", color.RedString("Only one of the following sections can be used:"), conflict)
+			}
+		}
+
+		// Return a general error message
+		return fmt.Errorf("found invalid configurations in config file: %s", v.ConfigFileUsed())
```

Comment:
we should point the user to a sample config file in documentation here. add a TODO.